src/main/java/games/strategy/triplea/ui/BattleDisplay.java

```diff
@@ -425,13 +425,18 @@ private Territory getRetreatInternal(final String message, final Collection<Terr
       // if you have eliminated the impossible, whatever remains, no matter
       // how improbable, must be the truth
       // retreat
-      final RetreatComponent comp = new RetreatComponent(possible);
-      final int option = JOptionPane.showConfirmDialog(BattleDisplay.this, comp, message,
-          JOptionPane.OK_CANCEL_OPTION, JOptionPane.PLAIN_MESSAGE, null);
-      if (option == JOptionPane.OK_OPTION) {
-        if (comp.getSelection() != null) {
-          retreatTo[0] = comp.getSelection();
-          latch.countDown();
+      if (possible.size() == 1) {
+        retreatTo[0] = possible.iterator().next();
+        latch.countDown();
```

Comment:
So a potential problem with this is that you may not want to retreat after seeing you only have 1 option. So consider that I attacked with some units and that I conquered a province adjacent to it already this turn. The current battle is going poorly and I'm thinking that I'll retreat to the newly conquered province and forget that I can't. Then the retreat options pop-up and I realize that I can only retreat to a province that I don't want to retreat to.

I think you could probably skip the retreat options for air units that only have 1 option (this is probably the main case you want to avoid prompting the user). Land units on the other hand I think you'd need to add information on possible retreat locations to the first prompt in order to remove selection prompt if there is only 1 option.